Panic when running debug zip

**Describe the problem**

Panic when running debug zip

**To Reproduce**

This is from https://github.com/cockroachdb/cockroach/issues/35357

```
  debug/nodes/3/details: rpc error: code = Unknown desc = initial connection heartbeat failed: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial tcp 172.31.23.235:26257: connect: connection refused"
  debug/nodes/3/details
  debug/nodes/3/gossip: rpc error: code = Unknown desc = initial connection heartbeat failed: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial tcp 172.31.23.235:26257: connect: connection refused"
  debug/nodes/3/gossip
E190304 19:52:43.125150 1 util/log/crash_reporting.go:202  a panic has occurred!
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x2531fef]

goroutine 1 [running]:
github.com/cockroachdb/cockroach/pkg/util/log.RecoverAndReportPanic(0x3a16600, 0xc0000520a0, 0xc000645300)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/crash_reporting.go:100 +0x9f
panic(0x2e6a280, 0x56c41d0)
	/usr/local/go/src/runtime/panic.go:513 +0x1b9
github.com/cockroachdb/cockroach/pkg/cli.runDebugZip.func10(0x3a16640, 0xc005359740, 0x2540be400, 0x3a16640)
	/go/src/github.com/cockroachdb/cockroach/pkg/cli/zip.go:279 +0xdf
github.com/cockroachdb/cockroach/pkg/util/contextutil.RunWithTimeout(0x3a16640, 0xc005359740, 0x3263dcb, 0xe, 0x2540be400, 0xc005568b28, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/contextutil/context.go:90 +0xb5
github.com/cockroachdb/cockroach/pkg/cli.runDebugZip(0x56edae0, 0xc000662f00, 0x1, 0x3, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/cli/zip.go:275 +0xbb9
github.com/cockroachdb/cockroach/pkg/cli.MaybeDecorateGRPCError.func1(0x56edae0, 0xc000662f00, 0x1, 0x3, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/cli/error.go:69 +0x6b
github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra.(*Command).execute(0x56edae0, 0xc000662ed0, 0x3, 0x3, 0x56edae0, 0xc000662ed0)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra/command.go:762 +0x473
github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x56ea900, 0x8, 0x0, 0xc000b13ed8)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra/command.go:852 +0x2fd
github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra.(*Command).Execute(0x56ea900, 0x18, 0xc000b13ef0)
	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra/command.go:800 +0x2b
github.com/cockroachdb/cockroach/pkg/cli.Run(0xc00004c1f0, 0x5, 0x5, 0xc0000520a0, 0xc000645300)
	/go/src/github.com/cockroachdb/cockroach/pkg/cli/cli.go:195 +0x65
github.com/cockroachdb/cockroach/pkg/cli.Main()
	/go/src/github.com/cockroachdb/cockroach/pkg/cli/cli.go:70 +0x222
main.main()
	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/cockroach/main.go:30 +0x20
Error:  exit status 2
```

--- a/pkg/cli/zip.go
+++ b/pkg/cli/zip.go
@@ -275,7 +275,7 @@ func runDebugZip(cmd *cobra.Command, args []string) error {
 				err = contextutil.RunWithTimeout(baseCtx, "request stacks", timeout,
 					func(ctx context.Context) error {
 						stacks, err := status.Stacks(ctx, &serverpb.StacksRequest{NodeId: id})
-						if err != nil {
+						if err == nil {
 							stacksData = stacks.Data
 						}
 						return err
@@ -291,7 +291,9 @@ func runDebugZip(cmd *cobra.Command, args []string) error {
 							NodeId: id,
 							Type:   serverpb.ProfileRequest_HEAP,
 						})
-						heapData = heap.Data
+						if err == nil {
+							heapData = heap.Data
+						}
 						return err
 					})
 				if err := z.createRawOrError(prefix+"/heap", heapData, err); err != nil {

--- a/pkg/cmd/roachtest/debug.go
+++ b/pkg/cmd/roachtest/debug.go
@@ -66,8 +66,17 @@ func registerDebug(r *registry) {
 
 		waitForFullReplication(t, db)
 
-		// Kill first nodes-1 nodes.
-		for i := 1; i < nodes; i++ {
+		// Kill the first node and verify debug zip can be generated with one node
+		// down.
+		c.Stop(ctx, c.Node(1))
+
+		// Run debug zip command against the second node.
+		if err := debugExtractExist(2, "output.zip"); err != nil {
+			t.Fatal(err)
+		}
+
+		// Kill all but the last node.
+		for i := 2; i < nodes; i++ {
 			c.Stop(ctx, c.Node(i))
 		}
 

